diesel/src/pg/connection/mod.rs

```diff
@@ -25,6 +25,92 @@ use crate::RunQueryDsl;
 /// The connection string expected by `PgConnection::establish`
 /// should be a PostgreSQL connection string, as documented at
 /// <https://www.postgresql.org/docs/9.4/static/libpq-connect.html#LIBPQ-CONNSTRING>
+///
+/// # Supported loading model implementations
+///
+/// * [`DefaultLoadingMode`]
+/// * [`PgRowByRowLoadingMode`]
+///
+/// If you are unsure which loading mode is the correct one for your application,
+/// you likely want to use the `DefaultLoadingMode` as that one offers
+/// generally better performance.
+///
+/// Due to the fact that `PgConnection` supports multiple loading modes
+/// it is **required** to always specify the used loading mode
+/// when calling [`RunQueryDsl::load_iter`] or [`LoadConnection::load`].
+///
+/// ## `DefaultLoadingMode`
+///
+/// By using this mode `PgConnection` defaults to loading all response values at **once**
+/// and only performs deserialization afterward for the `DefaultLoadingMode`.
+/// Generally this mode will be more performant as it.
+///
+/// This loading mode allows users to perform hold more than one iterator at once using
+/// the same connection:
+/// ```rust
+/// # include!("../../doctest_setup.rs");
+/// #
+/// # fn main() {
+/// #     run_test().unwrap();
+/// # }
+/// #
+/// # fn run_test() -> QueryResult<()> {
+/// #     use schema::users;
+/// #     let connection = &mut establish_connection();
+/// use diesel::connection::DefaultLoadingMode;
+///
+/// let iter1 = users::table.load_iter::<(i32, String), DefaultLoadingMode>(connection)?;
+/// let iter2 = users::table.load_iter::<(i32, String), DefaultLoadingMode>(connection)?;
+///
+/// for r in iter1 {
+///     let (id, name) = r?;
+///     println!("Id: {} Name: {}", id, name);
+/// }
+///
+/// for r in iter2 {
+///     let (id, name) = r?;
+///     println!("Id: {} Name: {}", id, name);
+/// }
```

Comment:
This currently fails to compile for whatever reason. I need to figure out where the additional lifetime comes from.